timber/src/commonMain/kotlin/timber/log/Timber.kt

```diff
@@ -0,0 +1,122 @@
+        open fun i(t: Throwable?, message: String?, vararg args: Any?)
+
+        /** Log an info exception. */
+        open fun i(t: Throwable?)
+
+        /** Log a warning message with optional format args. */
+        open fun w(message: String?, vararg args: Any?)
+
+        /** Log a warning exception and a message with optional format args. */
+        open fun w(t: Throwable?, message: String?, vararg args: Any?)
+
+        /** Log a warning exception. */
+        open fun w(t: Throwable?)
+
+        /** Log an error message with optional format args. */
+        open fun e(message: String?, vararg args: Any?)
+
+        /** Log an error exception and a message with optional format args. */
+        open fun e(t: Throwable?, message: String?, vararg args: Any?)
+
+        /** Log an error exception. */
+        open fun e(t: Throwable?)
+
+        /** Log an assert message with optional format args. */
+        open fun wtf(message: String?, vararg args: Any?)
+
+        /** Log an assert exception and a message with optional format args. */
+        open fun wtf(t: Throwable?, message: String?, vararg args: Any?)
+
+        /** Log an assert exception. */
+        open fun wtf(t: Throwable?)
+
+        /** Log at `priority` a message with optional format args. */
+        open fun log(priority: Int, message: String?, vararg args: Any?)
+
+        /** Log at `priority` an exception and a message with optional format args. */
+        open fun log(priority: Int, t: Throwable?, message: String?, vararg args: Any?)
+
+        /** Log at `priority` an exception. */
+        open fun log(priority: Int, t: Throwable?)
+
+        /** Return whether a message at `priority` or `tag` should be logged. */
+        protected open fun isLoggable(tag: String?, priority: Int): Boolean
+
+        /** Formats a log message with optional arguments. */
+        protected open fun formatMessage(message: String, args: Array<out Any?>): String
+
+        /**
+         * Write a log message to its destination. Called for all level-specific methods by default.
+         *
+         * @param priority Log level. See [Log] for constants.
+         * @param tag Explicit or inferred tag. May be `null`.
+         * @param message Formatted log message.
+         * @param t Accompanying exceptions. May be `null`.
+         */
+        protected abstract fun log(priority: Int, tag: String?, message: String, t: Throwable?)
+    }
+
+    companion object Forest: Tree {
+        /**
+         * A view into Timber's planted trees as a tree itself. This can be used for injecting a logger
+         * instance rather than using static methods or to facilitate testing.
+         */
+        @Suppress(
+            "NOTHING_TO_INLINE", // Kotlin users should reference `Tree.Forest` directly.
+            "NON_FINAL_MEMBER_IN_OBJECT" // For japicmp check.
+        )
+        open inline fun asTree(): Tree
```

Comment:
This doesn't need to be propagated to the common source set. It only exists for compatibility for the old Java implementation.
```suggestion
```